src/htm/encoders/RandomDistributedScalarEncoder.hpp

```diff
@@ -109,16 +109,47 @@ struct RDSE_Parameters
  * To inspect this run:
  * $ python -m htm.examples.encoders.rdse --help
  */
-class RandomDistributedScalarEncoder : public BaseEncoder<Real64>
+class RandomDistributedScalarEncoder : public GenericEncoder
 {
 public:
-  RandomDistributedScalarEncoder() {}
-  RandomDistributedScalarEncoder( const RDSE_Parameters &parameters );
-  void initialize( const RDSE_Parameters &parameters );
+  RandomDistributedScalarEncoder(); // for use by EncoderRegion.
+  RandomDistributedScalarEncoder( const RDSE_Parameters &params );  // for direct access
+  RandomDistributedScalarEncoder(ArWrapper &ar);                    // for Cereal load_ar( )
+
+
+  void initialize(BaseParameters *params) override {  // for use by generic EncoderRegion
+    initialize(*(RDSE_Parameters *)params);
+  }
+  void initialize(const RDSE_Parameters &params);
+
+  std::string getName() const override { return "RDSE"; }
+
+  // A discription of the fields in the RDSE_Parameters structure.
+  ParameterDescriptor getDescriptor() override {
```

Comment:
> Ok if the override is optional, I wouldn't be too happy if it were required for each new enc

If we are to use an encoder as a plugin to networkAPI with (Spec) then yes, all encoders will be required to provide this information.  Otherwise, networkAPI does not know how to call it.